### DISCUSSION TITLE
How to send @request.auth from server component using fetch?

### DISCUSSION THREAD

**Discussion by gaganbiswas** [pocketbase/pocketbase#4246] [Q&A] [answered]
I have the following List rule in my database: 
`@request.auth.id != "" && @request.auth.role = 'lib-admin'`

From my server component I am fetching the data from the collection using fetch:
```js
const res = await fetch("<POCKETBASE_URL>/api/collections/books/records", {
method: 'get', 
headers: {
//some headers
});

const data = await res.json()
```
How shall I send the user id of the current user from here. Shall I send the user token as authorization token in the header, and any alternatives.

P.S. I have a custom Pocketbase client, which basically stories the user data and token after authentication in the cookies and authStore in the browser/localStorage is empty.

**Comment by ganigeorgiev** [comment, marked as answer]
`@request.auth` will be populated if you send a valid `Authorization:TOKEN` header with the request.

**Reply by gaganbiswas** [reply]
Works like a charm. Thanks!
